aodncore/util/wfs.py

```diff
@@ -0,0 +1,118 @@
+import json
+from collections import OrderedDict
+
+from owslib.etree import etree
+from owslib.fes import PropertyIsEqualTo
+from owslib.wfs import WebFeatureService
+
+from ..util import IndexedSet
+
+__all__ = [
+    'WfsBroker',
+    'get_filter_for_file_url',
+    'ogc_filter_to_string'
+]
+
+
+def ogc_filter_to_string(ogc_filter):
+    """Convert an OGC filter object into it's XML string representation
+
+    :param ogc_filter: OGC filter object
+    :return: XML string
+    """
+    return etree.tostring(ogc_filter.toXML()).decode('utf-8')
+
+
+def get_filter_for_file_url(file_url, property_name='url'):
+    """Return OGC filter XML to query for a single file_url
+
+    :param file_url: URL string
+    :param property_name: URL property name to filter on
+    :return: OGC XML filter string
+    """
+    file_url_filter = PropertyIsEqualTo(propertyname=property_name, literal=file_url)
+    return ogc_filter_to_string(file_url_filter)
+
+
+class WfsBroker(object):
+    """Simple higher level interface to a WebFeatureService instance, to provide common helper methods and standardise
+    response handling around JSON
+    """
+
+    # The *first* matching property name found by the WebFeatureService.get_schema method will be considered to be the
+    # "url" property for a given layer. Accordingly, this should be ordered with highest priority name first.
+    url_propertyname_candidates = ('file_url', 'url')
+
+    def __init__(self, wfs_url, version='1.0.0'):
+        self._wfs = WebFeatureService(wfs_url, version=version)
+
+    @property
+    def wfs(self):
+        """Read-only property to access the instantiated WebFeatureService object directly
+
+        :return: WebFeatureService instance
+        """
+        return self._wfs
+
+    def getfeature_dict(self, **kwargs):
```

Comment:
A couple of comments after playing with this...

Although `WebFeatureService.getfeature` does seem to allow `typename=None` _if_ `featureid` is provided, I can't imagine any use case for this here, so for us I think `typename`  should be a required (positional) arg.

Also `getfeature` expects a _list_ of typenames, but we'll almost always want to query just one, so it would be nice to allow for the `typename` argument to be a string.

Reply:
E.g. something like this: https://github.com/aodn/python-aodncore/pull/139